tools/devops/automation/templates/governance/stage.yml

```diff
@@ -0,0 +1,95 @@
+# contains the stage used to run all governance related jobs
+
+parameters:
+- name: isPR
+  type: boolean
+
+- name: repositoryAlias
+  type: string
+  default: self
+
+- name: commit
+  type: string
+  default: HEAD
+
+- name: stageDisplayNamePrefix
+  type: string
+  default: ''
+
+stages:
+- stage: governance_checks
+  displayName: '${{ parameters.stageDisplayNamePrefix }}Governance Checks'
+  dependsOn: [ configure_build ]
+
+  jobs:
+  - job: apiscan
+    displayName: 'APIScan:' 
+    pool:
+      vmImage: windows-latest
+
+    variables:
+      INCLUDE_DOTNET_IOS: $[ stageDependencies.configure_build.configure.outputs['configure_platforms.INCLUDE_DOTNET_IOS'] ]
+      INCLUDE_DOTNET_TVOS: $[ stageDependencies.configure_build.configure.outputs['configure_platforms.INCLUDE_DOTNET_TVOS'] ]
+      INCLUDE_DOTNET_MACCATALYST: $[ stageDependencies.configure_build.configure.outputs['configure_platforms.INCLUDE_DOTNET_MACCATALYST'] ]
+      INCLUDE_DOTNET_MACOS: $[ stageDependencies.configure_build.configure.outputs['configure_platforms.INCLUDE_DOTNET_MACOS'] ]
+
+    strategy:
+      matrix:
+        iOS:
+          INCLUDE: $(INCLUDE_DOTNET_IOS)
+          PLATFORM: iOS
+          SECRET: $(ApiScanSecretiOS)
+          VERSION: 17.0
+        tvOS:
+          INCLUDE: $(INCLUDE_DOTNET_TVOS)
+          PLATFORM: tvOS
+          SECRET: $(ApiScanSecretiOS)
+          VERSION: 17.0
+        macCatalyst:
+          INCLUDE: $(INCLUDE_DOTNET_MACCATALYST)
+          PLATFORM: macCatalyst
+          SECRET: $(ApiScanSecretiOS)
+          VERSION: 17.0
+        macOS:
+          INCLUDE: $(INCLUDE_DOTNET_MACOS)
+          PLATFORM: macOS
+          SECRET: $(ApiScanSecretmacOS)
+          VERSION: 14.0
```

Comment:
I'm not very fond of hardcoding the list of platforms in yet another place (visionOS...), so would it be possible to generate this somehow? The DOTNET_PLATFORMS variable (`stageDependencies.configure_build.configure.outputs['configure_platforms.DOTNET_PLATFORMS']`) should contain the list of enabled platforms.

Reply:
lets see if I can do that, but there are certain problem with using a runtime variable in a template.

Reply:
@rolfbjarne the change you are asking, as we have everything is impossible. With he yaml template synstax we have 3 diff stages:

- `${{ }}` : This is a template expression. It gets executed BEFORE anything runs and expands the data.
- `$[]`: Expanded at runtime.

A template cannot use a value that is generated and runtime. If we really wanted to get to the point in which we can generate things like the one you suggest, we should do a two step, in which we generate a yaml file with the variables/parameters needed by the templates and then generated. Else we can move to use template parameter form the pipeline and push them down. It will mean we need to update those params in a single place.

AZP does not give us many options. If we want to go that route, we need to move the new platforms param I added to the entry level yaml for both ci and pr.

Reply:
You can generate the matrix, this is what the test splitting PR does:

https://github.com/mcumming/xamarin-macios/blob/79b9eaa0ea91e6a9fbc6707d4b8ee36930736ba3/tools/devops/automation/templates/tests/stage.yml#L123